.github/workflows/release.yml

```
@@ -29,58 +28,40 @@ jobs:
           - variant: bookworm
             is_default_variant: true
     steps:
-      - uses: actions/checkout@v3
-      - uses: docker/setup-qemu-action@v2
-        with:
-          platforms: ${{ env.PLATFORMS }}
-      - uses: docker/setup-buildx-action@v2
-      - uses: docker/login-action@v2
+      - uses: actions/checkout@v4.1.1
+      - uses: docker/login-action@v3
         with:
           registry: ghcr.io
           username: ${{ github.repository_owner }}
           password: ${{ secrets.GITHUB_TOKEN }}
       - name: "Python + Variant: ${{ env.POETRY_VERSION }}-python-${{ matrix.python }}-${{ matrix.variant }}"
-        uses: docker/build-push-action@v3.1.0
-        with:
-          push: true
-          tags: ${{ env.GHCR_IMAGE_REPOSITORY }}:${{ env.POETRY_VERSION }}-python-${{ matrix.python }}-${{ matrix.variant }}
-          cache-from: type=gha
-          cache-to: type=gha,mode=max
-          platforms: ${{ env.PLATFORMS }}
-          build-args: |
-            POETRY_VERSION=${{ env.POETRY_VERSION }}
-            PYTHON_VERSION=${{ matrix.python }}
-            VARIANT=${{ matrix.variant }}
+        run: |
+          ORIGINAL_TAG="${{ env.GHCR_IMAGE_REPOSITORY }}:${{ github.sha }}-${{ matrix.python }}-${{ matrix.variant }}"
+          NEW_TAG="${{ env.GHCR_IMAGE_REPOSITORY }}:${{ env.POETRY_VERSION }}-python-${{ matrix.python }}-${{ matrix.variant }}"
+          docker pull $ORIGINAL_TAG
+          docker tag $ORIGINAL_TAG $NEW_TAG
+          docker push $NEW_TAG
```

I suggest we use semantic variable names and reuse what we can

```suggestion
          TAG_SUFFIX="${{ matrix.python }}-${{ matrix.variant }}"
          GIT_SHA_TAG="${{ env.GHCR_IMAGE_REPOSITORY }}:${{ github.sha }}-${TAG_SUFFIX}"
          RELEASE_TAG="${{ env.GHCR_IMAGE_REPOSITORY }}:${{ env.POETRY_VERSION }}-python-${TAG_SUFFIX}"
          docker pull $GIT_SHA_TAG
          docker tag $GIT_SHA_TAG $RELEASE_TAG
          docker push $RELEASE_TAG
```